### PR TITLE
upgrade baa, fst-writer and fraction dependencies

### PR DESCRIPTION
**Pull request by ekiwi**

### CODE REVIEW

File: cider/Cargo.toml [hunk 1]
```diff
@@ -21,10 +21,9 @@ pest_derive.workspace = true
 pest_consume.workspace = true
 argh.workspace = true
 owo-colors = "^3.5"
-bitvec = "1.0"
```
Comment (ekiwi) [inline cider/Cargo.toml:24]: this seemed unused so I removed it

File: cider/Cargo.toml [hunk 2]
```diff
@@ -33,7 +32,6 @@ ahash = "0.8.3"
 num-bigint = "0.4.6"
 num-traits = "0.2.19"
 
-once_cell = "1.9.0"
```
Comment (ekiwi) [inline cider/Cargo.toml:36]: this seemed unused so I removed it

File: cider/Cargo.toml [hunk 3]
```diff
@@ -42,8 +40,8 @@ calyx-opt = { path = "../calyx-opt" }
 calyx-frontend = { path = "../calyx-frontend" }
 
 ciborium = "0.2.2"
-baa = { version = "0.14.0", features = ["bigint", "serde1", "fraction1"] }
-fst-writer = "0.2.1"
+baa.workspace = true
```
Comment (ekiwi) [inline cider/Cargo.toml:43]: this unifies the baa version between `cider` and `cider-dap`

**Review by EclecticGriffin — APPROVED** [review]
Seems good to me!

**Pull request merged**